Latest build fails to compile, possibly some update related to unit tests judging from error message

- OpenCV => trunk, latest
- Operating System / Platform => Ubuntu 16.04
- Compiler => gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4)

##### Steps to reproduce

``````
cd opencv && mkdir build && cd build
cmake ..
make
``````

Error message

``````
[ 26%] Building CXX object modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/perf/perf_arithm.cpp.o
In file included from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:55:0:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:70:103: error: template argument 2 is invalid
     #define DEF_PARAM_TEST(name, ...) typedef ::perf::TestBaseWithParam< std::tr1::tuple< __VA_ARGS__ > > name
                                                                                                       ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:110:1: note: in expansion of macro ‘DEF_PARAM_TEST’
 DEF_PARAM_TEST(Sz_Flags, cv::Size, DftFlags);
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:70:105: error: template argument 1 is invalid
     #define DEF_PARAM_TEST(name, ...) typedef ::perf::TestBaseWithParam< std::tr1::tuple< __VA_ARGS__ > > name
                                                                                                         ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:110:1: note: in expansion of macro ‘DEF_PARAM_TEST’
 DEF_PARAM_TEST(Sz_Flags, cv::Size, DftFlags);
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:605:45: error: expected class-name before ‘{’ token
     class fixture##_##name : public fixture {\
                                             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘class Sz_Flags_MulSpectrums’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:605:11: note: in definition of macro ‘PERF_TEST_P’
     class fixture##_##name : public fixture {\
           ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: base class ‘class Sz_Flags_MulSpectrums’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:9045:3: note: in definition of macro ‘GTEST_TEST_CLASS_NAME_’
   test_case_name##_##test_name##_Test
   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘class Sz_Flags_MulSpectrums_MulSpectrums_Test’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:9045:3: note: in definition of macro ‘GTEST_TEST_CLASS_NAME_’
   test_case_name##_##test_name##_Test
   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11646:40: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
   typedef typename TestCase::ParamType ParamType;
                                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11648:60: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
   typedef ParamGenerator<ParamType>(GeneratorCreationFunc)();
                                                            ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11649:54: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
   typedef typename ParamNameGenFunc<ParamType>::Type ParamNameGeneratorFunc;
                                                      ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11665:8: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
   void AddTestPattern(const char* test_case_name,
        ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:29:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In static member function ‘static int Sz_Flags_MulSpectrums_MulSpectrums_Test::AddToRegistry()’:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_ext.hpp:79:41: error: ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>’ has no member named ‘AddTestPattern’
                   __FILE__, __LINE__))->AddTestPattern(\
                                         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘class testing::internal::TestMetaFactory<Sz_Flags_MulSpectrums_MulSpectrums_Test>’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11587:7: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums_MulSpectrums_Test’
 class TestMetaFactory
       ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11590:40: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums_MulSpectrums_Test’
   typedef typename TestCase::ParamType ParamType;
                                        ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In member function ‘virtual void Sz_Flags_MulSpectrums_MulSpectrums_Test::Body()’:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:62: error: ‘RunPerfTestBody’ was not declared in this scope
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:39: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                       ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:39: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                       ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:64: error: template argument 1 is invalid
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                                                ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In function ‘int gtest_Sz_Flags_MulSpectrums_EvalGenerator_()’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:114:31: error: reference to ‘DftFlags’ is ambiguous
                     Values(0, DftFlags(cv::DFT_ROWS))))
                               ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18247:66: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       gtest_##prefix##test_case_name##_EvalGenerator_() { return generator; } \
                                                                  ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:108:10: note: candidates are: struct {anonymous}::DftFlags
 CV_FLAGS(DftFlags, 0, cv::DFT_INVERSE, cv::DFT_SCALE, cv::DFT_ROWS, cv::DFT_COMPLEX_OUTPUT, cv::DFT_REAL_OUTPUT)
          ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:127:12: note: in definition of macro ‘CV_FLAGS’
     struct class_name {                                                                 \
            ^
In file included from /home/jasjuang/opencv/modules/core/include/opencv2/core.hpp:54:0,
                 from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:41,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/core/include/opencv2/core/base.hpp:214:6: note:                 enum cv::DftFlags
 enum DftFlags {
      ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:63: error: template argument 1 is invalid
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                                               ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In function ‘std::__cxx11::string gtest_Sz_Flags_MulSpectrums_EvalGenerateName_(const int&)’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18250:49: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
     return ::testing::internal::GetParamNameGen<test_case_name::ParamType> \
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_MulSpectrums’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18250:49: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
     return ::testing::internal::GetParamNameGen<test_case_name::ParamType> \
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: no matching function for call to ‘GetParamNameGen()’
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11531:21: note: candidate: template<class ParamType, class ParamNameGenFunctor> ParamNameGenFunctor testing::internal::GetParamNameGen(ParamNameGenFunctor)
 ParamNameGenFunctor GetParamNameGen(ParamNameGenFunctor func) {
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11531:21: note:   template argument deduction/substitution failed:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: template argument 1 is invalid
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11541:45: note: candidate: template<class ParamType> typename testing::internal::ParamNameGenFunc<ParamType>::Type* testing::internal::GetParamNameGen()
 typename ParamNameGenFunc<ParamType>::Type *GetParamNameGen() {
                                             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11541:45: note:   template argument deduction/substitution failed:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: template argument 1 is invalid
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:41: error: ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>’ has no member named ‘AddTestCaseInstantiation’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, MulSpectrums,
 ^
In file included from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:55:0:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In member function ‘virtual void Sz_Flags_MulSpectrums::PerfTestBody()’:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:68:54: error: ‘GetParam’ was not declared in this scope
     #define GET_PARAM(k) std::tr1::get< k >(GetParam())
                                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:116:27: note: in expansion of macro ‘GET_PARAM’
     const cv::Size size = GET_PARAM(0);
                           ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:121:5: error: ‘declare’ was not declared in this scope
     declare.in(a, b, WARMUP_RNG);
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:121:22: error: ‘WARMUP_RNG’ was not declared in this scope
     declare.in(a, b, WARMUP_RNG);
                      ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:33: error: ‘next’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:129:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::mulSpectrums(d_a, d_b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:49: error: ‘startTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:129:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::mulSpectrums(d_a, d_b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:62: error: ‘stopTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:129:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::mulSpectrums(d_a, d_b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:206:94: error: no matching function for call to ‘perf::Regression::add(Sz_Flags_MulSpectrums*, const char [8], cv::Mat&, double, perf::ERROR_TYPE)’
 #define SANITY_CHECK(array, ...) ::perf::Regression::add(this, #array, array , ## __VA_ARGS__)
                                                                                              ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:85:13: note: in expansion of macro ‘SANITY_CHECK’
             SANITY_CHECK(gpu_##mat, ## __VA_ARGS__); \
             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:131:9: note: in expansion of macro ‘CUDA_SANITY_CHECK’
         CUDA_SANITY_CHECK(dst, 1e-6, ERROR_RELATIVE);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note: candidate: static perf::Regression& perf::Regression::add(perf::TestBase*, const string&, cv::InputArray, double, perf::ERROR_TYPE)
     static Regression& add(TestBase* test, const std::string& name, cv::InputArray array, double eps = DBL_EPSILON, ERROR_TYPE err = ERROR_ABSOLUTE);
                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note:   no known conversion for argument 1 from ‘Sz_Flags_MulSpectrums*’ to ‘perf::TestBase*’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:33: error: ‘next’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:137:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::mulSpectrums(a, b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:49: error: ‘startTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:137:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::mulSpectrums(a, b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:62: error: ‘stopTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:137:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::mulSpectrums(a, b, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:206:94: error: no matching function for call to ‘perf::Regression::add(Sz_Flags_MulSpectrums*, const char [8], cv::Mat&)’
 #define SANITY_CHECK(array, ...) ::perf::Regression::add(this, #array, array , ## __VA_ARGS__)
                                                                                              ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:91:13: note: in expansion of macro ‘SANITY_CHECK’
             SANITY_CHECK(cpu_##mat, ## __VA_ARGS__); \
             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:139:9: note: in expansion of macro ‘CPU_SANITY_CHECK’
         CPU_SANITY_CHECK(dst);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note: candidate: static perf::Regression& perf::Regression::add(perf::TestBase*, const string&, cv::InputArray, double, perf::ERROR_TYPE)
     static Regression& add(TestBase* test, const std::string& name, cv::InputArray array, double eps = DBL_EPSILON, ERROR_TYPE err = ERROR_ABSOLUTE);
                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note:   no known conversion for argument 1 from ‘Sz_Flags_MulSpectrums*’ to ‘perf::TestBase*’
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:605:45: error: expected class-name before ‘{’ token
     class fixture##_##name : public fixture {\
                                             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘class Sz_Flags_Dft’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:605:11: note: in definition of macro ‘PERF_TEST_P’
     class fixture##_##name : public fixture {\
           ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: base class ‘class Sz_Flags_Dft’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:9045:3: note: in definition of macro ‘GTEST_TEST_CLASS_NAME_’
   test_case_name##_##test_name##_Test
   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘class Sz_Flags_Dft_Dft_Test’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:9045:3: note: in definition of macro ‘GTEST_TEST_CLASS_NAME_’
   test_case_name##_##test_name##_Test
   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11646:40: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
   typedef typename TestCase::ParamType ParamType;
                                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11648:60: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
   typedef ParamGenerator<ParamType>(GeneratorCreationFunc)();
                                                            ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11649:54: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
   typedef typename ParamNameGenFunc<ParamType>::Type ParamNameGeneratorFunc;
                                                      ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11665:8: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
   void AddTestPattern(const char* test_case_name,
        ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:29:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In static member function ‘static int Sz_Flags_Dft_Dft_Test::AddToRegistry()’:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_ext.hpp:79:41: error: ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>’ has no member named ‘AddTestPattern’
                   __FILE__, __LINE__))->AddTestPattern(\
                                         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:5: note: in expansion of macro ‘TEST_P’
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘class testing::internal::TestMetaFactory<Sz_Flags_Dft_Dft_Test>’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11587:7: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft_Dft_Test’
 class TestMetaFactory
       ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11590:40: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft_Dft_Test’
   typedef typename TestCase::ParamType ParamType;
                                        ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In member function ‘virtual void Sz_Flags_Dft_Dft_Test::Body()’:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:611:62: error: ‘RunPerfTestBody’ was not declared in this scope
     TEST_P(fixture##_##name, name /*perf*/){ RunPerfTestBody(); }\
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:39: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                       ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:39: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                       ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18246:64: error: template argument 1 is invalid
   ::testing::internal::ParamGenerator<test_case_name::ParamType> \
                                                                ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In function ‘int gtest_Sz_Flags_Dft_EvalGenerator_()’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:178:31: error: reference to ‘DftFlags’ is ambiguous
                     Values(0, DftFlags(cv::DFT_ROWS), DftFlags(cv::DFT_INVERSE))))
                               ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18247:66: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       gtest_##prefix##test_case_name##_EvalGenerator_() { return generator; } \
                                                                  ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:108:10: note: candidates are: struct {anonymous}::DftFlags
 CV_FLAGS(DftFlags, 0, cv::DFT_INVERSE, cv::DFT_SCALE, cv::DFT_ROWS, cv::DFT_COMPLEX_OUTPUT, cv::DFT_REAL_OUTPUT)
          ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:127:12: note: in definition of macro ‘CV_FLAGS’
     struct class_name {                                                                 \
            ^
In file included from /home/jasjuang/opencv/modules/core/include/opencv2/core.hpp:54:0,
                 from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:41,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/core/include/opencv2/core/base.hpp:214:6: note:                 enum cv::DftFlags
 enum DftFlags {
      ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:178:55: error: reference to ‘DftFlags’ is ambiguous
                     Values(0, DftFlags(cv::DFT_ROWS), DftFlags(cv::DFT_INVERSE))))
                                                       ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18247:66: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       gtest_##prefix##test_case_name##_EvalGenerator_() { return generator; } \
                                                                  ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:108:10: note: candidates are: struct {anonymous}::DftFlags
 CV_FLAGS(DftFlags, 0, cv::DFT_INVERSE, cv::DFT_SCALE, cv::DFT_ROWS, cv::DFT_COMPLEX_OUTPUT, cv::DFT_REAL_OUTPUT)
          ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:127:12: note: in definition of macro ‘CV_FLAGS’
     struct class_name {                                                                 \
            ^
In file included from /home/jasjuang/opencv/modules/core/include/opencv2/core.hpp:54:0,
                 from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:41,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/core/include/opencv2/core/base.hpp:214:6: note:                 enum cv::DftFlags
 enum DftFlags {
      ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18249:63: error: template argument 1 is invalid
       const ::testing::TestParamInfo<test_case_name::ParamType>& info) { \
                                                               ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In function ‘std::__cxx11::string gtest_Sz_Flags_Dft_EvalGenerateName_(const int&)’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18250:49: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
     return ::testing::internal::GetParamNameGen<test_case_name::ParamType> \
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:13: error: ‘ParamType’ is not a member of ‘Sz_Flags_Dft’
 PERF_TEST_P(Sz_Flags, Dft,
             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18250:49: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
     return ::testing::internal::GetParamNameGen<test_case_name::ParamType> \
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: no matching function for call to ‘GetParamNameGen()’
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11531:21: note: candidate: template<class ParamType, class ParamNameGenFunctor> ParamNameGenFunctor testing::internal::GetParamNameGen(ParamNameGenFunctor)
 ParamNameGenFunctor GetParamNameGen(ParamNameGenFunctor func) {
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11531:21: note:   template argument deduction/substitution failed:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: template argument 1 is invalid
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11541:45: note: candidate: template<class ParamType> typename testing::internal::ParamNameGenFunc<ParamType>::Type* testing::internal::GetParamNameGen()
 typename ParamNameGenFunc<ParamType>::Type *GetParamNameGen() {
                                             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11541:45: note:   template argument deduction/substitution failed:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18251:21: error: template argument 1 is invalid
         (__VA_ARGS__)(info); \
                     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: At global scope:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:41: error: ‘class testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>’ has no member named ‘AddTestCaseInstantiation’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:612:5: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
     INSTANTIATE_TEST_CASE_P(/*none*/, fixture##_##name, params);\
     ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1: note: in expansion of macro ‘PERF_TEST_P’
 PERF_TEST_P(Sz_Flags, Dft,
 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp: In member function ‘virtual void Sz_Flags_Dft::PerfTestBody()’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:180:5: error: ‘declare’ was not declared in this scope
     declare.time(10.0);
     ^
In file included from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:55:0:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:68:54: error: ‘GetParam’ was not declared in this scope
     #define GET_PARAM(k) std::tr1::get< k >(GetParam())
                                                      ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:182:27: note: in expansion of macro ‘GET_PARAM’
     const cv::Size size = GET_PARAM(0);
                           ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:186:21: error: ‘WARMUP_RNG’ was not declared in this scope
     declare.in(src, WARMUP_RNG);
                     ^
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:596:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:33: error: ‘next’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:193:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::dft(d_src, dst, size, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:49: error: ‘startTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:193:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::dft(d_src, dst, size, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:62: error: ‘stopTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:193:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::cuda::dft(d_src, dst, size, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:206:94: error: no matching function for call to ‘perf::Regression::add(Sz_Flags_Dft*, const char [8], cv::Mat&, double, perf::ERROR_TYPE)’
 #define SANITY_CHECK(array, ...) ::perf::Regression::add(this, #array, array , ## __VA_ARGS__)
                                                                                              ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:85:13: note: in expansion of macro ‘SANITY_CHECK’
             SANITY_CHECK(gpu_##mat, ## __VA_ARGS__); \
             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:195:9: note: in expansion of macro ‘CUDA_SANITY_CHECK’
         CUDA_SANITY_CHECK(dst, 1e-6, ERROR_RELATIVE);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note: candidate: static perf::Regression& perf::Regression::add(perf::TestBase*, const string&, cv::InputArray, double, perf::ERROR_TYPE)
     static Regression& add(TestBase* test, const std::string& name, cv::InputArray array, double eps = DBL_EPSILON, ERROR_TYPE err = ERROR_ABSOLUTE);
                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note:   no known conversion for argument 1 from ‘Sz_Flags_Dft*’ to ‘perf::TestBase*’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:33: error: ‘next’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:201:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::dft(src, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:49: error: ‘startTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                 ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:201:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::dft(src, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:661:62: error: ‘stopTimer’ was not declared in this scope
 #define TEST_CYCLE() for(; next() && startTimer(); stopTimer())
                                                              ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:201:9: note: in expansion of macro ‘TEST_CYCLE’
         TEST_CYCLE() cv::dft(src, dst, flag);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:206:94: error: no matching function for call to ‘perf::Regression::add(Sz_Flags_Dft*, const char [8], cv::Mat&)’
 #define SANITY_CHECK(array, ...) ::perf::Regression::add(this, #array, array , ## __VA_ARGS__)
                                                                                              ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/cuda_perf.hpp:91:13: note: in expansion of macro ‘SANITY_CHECK’
             SANITY_CHECK(cpu_##mat, ## __VA_ARGS__); \
             ^
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:203:9: note: in expansion of macro ‘CPU_SANITY_CHECK’
         CPU_SANITY_CHECK(dst);
         ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note: candidate: static perf::Regression& perf::Regression::add(perf::TestBase*, const string&, cv::InputArray, double, perf::ERROR_TYPE)
     static Regression& add(TestBase* test, const std::string& name, cv::InputArray array, double eps = DBL_EPSILON, ERROR_TYPE err = ERROR_ABSOLUTE);
                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_perf.hpp:168:24: note:   no known conversion for argument 1 from ‘Sz_Flags_Dft*’ to ‘perf::TestBase*’
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/perf_precomp.hpp:54:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘struct testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo’:
/usr/include/c++/5/bits/stl_construct.h:151:15:   required from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, std::allocator<_T2>&) [with _ForwardIterator = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo>*; _Tp = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo>]’
/usr/include/c++/5/bits/stl_vector.h:424:22:   required from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo>; _Alloc = std::allocator<testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo> >]’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11653:60:   required from ‘testing::internal::ParameterizedTestCaseInfo<TestCase>::ParameterizedTestCaseInfo(const char*, testing::internal::CodeLocation) [with TestCase = Sz_Flags_MulSpectrums]’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11847:23:   required from ‘testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation) [with TestCase = Sz_Flags_MulSpectrums]’
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:112:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11747:5: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
     TestInfo(const char* a_test_case_base_name,
     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11756:55: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
     const scoped_ptr<TestMetaFactoryBase<ParamType> > test_meta_factory;
                                                       ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘struct testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo’:
/usr/include/c++/5/bits/stl_construct.h:151:15:   required from ‘void std::_Destroy(_ForwardIterator, _ForwardIterator, std::allocator<_T2>&) [with _ForwardIterator = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo>*; _Tp = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo>]’
/usr/include/c++/5/bits/stl_vector.h:424:22:   required from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo>; _Alloc = std::allocator<testing::internal::linked_ptr<testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo> >]’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11653:60:   required from ‘testing::internal::ParameterizedTestCaseInfo<TestCase>::ParameterizedTestCaseInfo(const char*, testing::internal::CodeLocation) [with TestCase = Sz_Flags_Dft]’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11847:23:   required from ‘testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation) [with TestCase = Sz_Flags_Dft]’
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:176:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11747:5: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
     TestInfo(const char* a_test_case_base_name,
     ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11756:55: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
     const scoped_ptr<TestMetaFactoryBase<ParamType> > test_meta_factory;
                                                       ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘void testing::internal::ParameterizedTestCaseInfo<TestCase>::RegisterTests() [with TestCase = Sz_Flags_Dft]’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:254:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11696:35: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
         ParamGenerator<ParamType> generator((*gen_it->generator)());
                                   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11708:59: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
         for (typename ParamGenerator<ParamType>::iterator param_it =
                                                           ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11708:59: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11696:35: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
         ParamGenerator<ParamType> generator((*gen_it->generator)());
                                   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11713:45: error: no type named ‘ParamType’ in ‘class Sz_Flags_Dft’
           std::string param_name = name_func(
                                             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11728:34: error: ‘SetUpTestCase’ is not a member of ‘Sz_Flags_Dft’
           MakeAndRegisterTestInfo(
                                  ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11728:34: error: ‘TearDownTestCase’ is not a member of ‘Sz_Flags_Dft’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11737:15: error: ‘struct testing::internal::ParameterizedTestCaseInfo<Sz_Flags_Dft>::TestInfo’ has no member named ‘test_meta_factory’
               test_info->test_meta_factory->CreateTestFactory(*param_it));
               ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h: In instantiation of ‘void testing::internal::ParameterizedTestCaseInfo<TestCase>::RegisterTests() [with TestCase = Sz_Flags_MulSpectrums]’:
/home/jasjuang/opencv/modules/cudaarithm/perf/perf_arithm.cpp:254:1:   required from here
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11696:35: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
         ParamGenerator<ParamType> generator((*gen_it->generator)());
                                   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11708:59: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
         for (typename ParamGenerator<ParamType>::iterator param_it =
                                                           ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11708:59: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11696:35: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
         ParamGenerator<ParamType> generator((*gen_it->generator)());
                                   ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11713:45: error: no type named ‘ParamType’ in ‘class Sz_Flags_MulSpectrums’
           std::string param_name = name_func(
                                             ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11728:34: error: ‘SetUpTestCase’ is not a member of ‘Sz_Flags_MulSpectrums’
           MakeAndRegisterTestInfo(
                                  ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11728:34: error: ‘TearDownTestCase’ is not a member of ‘Sz_Flags_MulSpectrums’
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11737:15: error: ‘struct testing::internal::ParameterizedTestCaseInfo<Sz_Flags_MulSpectrums>::TestInfo’ has no member named ‘test_meta_factory’
               test_info->test_meta_factory->CreateTestFactory(*param_it));
               ^
cc1plus: some warnings being treated as errors
modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/build.make:86: recipe for target 'modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/perf/perf_arithm.cpp.o' failed
make[2]: *** [modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/perf/perf_arithm.cpp.o] Error 1
CMakeFiles/Makefile2:1775: recipe for target 'modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/all' failed
make[1]: *** [modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
``````

I applied fix #7747 but there is still error. There is less error message though

``````
[ 23%] Building CXX object modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/test/test_core.cpp.o
In file included from /home/jasjuang/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/jasjuang/opencv/build/modules/cudaarithm/test_precomp.hpp:56:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18255:11: error: parse error in template argument list
           GetTestCasePatternHolder<test_case_name>(\
           ^
/home/jasjuang/opencv/modules/cudaarithm/test/test_core.cpp:353:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Arithm, LUT, testing::Combine(
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: no matching function for call to ‘testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char [4], testing::internal::CodeLocation)’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/test/test_core.cpp:353:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Arithm, LUT, testing::Combine(
 ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note: candidate: template<class TestCase> testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation)
   ParameterizedTestCaseInfo<TestCase>* GetTestCasePatternHolder(
                                        ^
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note:   template argument deduction/substitution failed:
/home/jasjuang/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: template argument 1 is invalid
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/jasjuang/opencv/modules/cudaarithm/test/test_core.cpp:353:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Arithm, LUT, testing::Combine(
 ^
modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/build.make:158: recipe for target 'modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/test/test_core.cpp.o' failed
make[2]: *** [modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/test/test_core.cpp.o] Error 1
CMakeFiles/Makefile2:2775: recipe for target 'modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/all' failed
make[1]: *** [modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
``````

Actually my bad, there is still compilation errors

``````
[ 36%] Building CXX object modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/test/test_filters.cpp.o
In file included from /home/hypevr/opencv/modules/ts/include/opencv2/ts.hpp:28:0,
                 from /home/hypevr/opencv/build/modules/cudafilters/test_precomp.hpp:54:
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18255:11: error: parse error in template argument list
           GetTestCasePatternHolder<test_case_name>(\
           ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:212:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Laplacian, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: no matching function for call to ‘testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char [10], testing::internal::CodeLocation)’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:212:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Laplacian, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note: candidate: template<class TestCase> testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation)
   ParameterizedTestCaseInfo<TestCase>* GetTestCasePatternHolder(
                                        ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note:   template argument deduction/substitution failed:
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: template argument 1 is invalid
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:212:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Laplacian, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18255:11: error: parse error in template argument list
           GetTestCasePatternHolder<test_case_name>(\
           ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:341:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Sobel, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: no matching function for call to ‘testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char [6], testing::internal::CodeLocation)’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:341:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Sobel, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note: candidate: template<class TestCase> testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation)
   ParameterizedTestCaseInfo<TestCase>* GetTestCasePatternHolder(
                                        ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note:   template argument deduction/substitution failed:
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: template argument 1 is invalid
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:341:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Sobel, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18255:11: error: parse error in template argument list
           GetTestCasePatternHolder<test_case_name>(\
           ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:406:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Scharr, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: no matching function for call to ‘testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char [7], testing::internal::CodeLocation)’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:406:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Scharr, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note: candidate: template<class TestCase> testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation)
   ParameterizedTestCaseInfo<TestCase>* GetTestCasePatternHolder(
                                        ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note:   template argument deduction/substitution failed:
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: template argument 1 is invalid
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:406:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, Scharr, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18255:11: error: parse error in template argument list
           GetTestCasePatternHolder<test_case_name>(\
           ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:467:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, GaussianBlur, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: no matching function for call to ‘testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char [13], testing::internal::CodeLocation)’
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:467:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, GaussianBlur, testing::Combine(
 ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note: candidate: template<class TestCase> testing::internal::ParameterizedTestCaseInfo<TestCase>* testing::internal::ParameterizedTestCaseRegistry::GetTestCasePatternHolder(const char*, testing::internal::CodeLocation)
   ParameterizedTestCaseInfo<TestCase>* GetTestCasePatternHolder(
                                        ^
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:11823:40: note:   template argument deduction/substitution failed:
/home/hypevr/opencv/modules/ts/include/opencv2/ts/ts_gtest.h:18258:38: error: template argument 1 is invalid
                   __FILE__, __LINE__))->AddTestCaseInstantiation(\
                                      ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:467:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
 INSTANTIATE_TEST_CASE_P(CUDA_Filters, GaussianBlur, testing::Combine(
 ^
/home/hypevr/opencv/modules/cudafilters/test/test_filters.cpp:65:13: warning: ‘cv::Mat {anonymous}::getInnerROI(cv::InputArray, int)’ defined but not used [-Wunused-function]
     cv::Mat getInnerROI(cv::InputArray m, int ksize)
             ^
modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/build.make:86: recipe for target 'modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/test/test_filters.cpp.o' failed
make[2]: *** [modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/test/test_filters.cpp.o] Error 1
CMakeFiles/Makefile2:5504: recipe for target 'modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/all' failed
make[1]: *** [modules/cudafilters/CMakeFiles/opencv_test_cudafilters.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
``````

--- a/modules/cudaarithm/perf/perf_arithm.cpp
+++ b/modules/cudaarithm/perf/perf_arithm.cpp
@@ -46,6 +46,8 @@ using namespace std;
 using namespace testing;
 using namespace perf;
 
+namespace { // workaround conflict with DftFlags
+
 //////////////////////////////////////////////////////////////////////
 // GEMM
 
@@ -252,3 +254,5 @@ PERF_TEST_P(Sz_KernelSz_Ccorr, Convolve,
         CPU_SANITY_CHECK(dst);
     }
 }
+
+} // namespace

--- a/modules/cudaarithm/test/test_core.cpp
+++ b/modules/cudaarithm/test/test_core.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 ////////////////////////////////////////////////////////////////////////////////
 // Merge
 
@@ -415,5 +417,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_Arithm, CopyMakeBorder, testing::Combine(
     testing::Values(Border(1), Border(10), Border(50)),
     ALL_BORDER_TYPES,
     WHOLE_SUBMAT));
+
+} //namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudafeatures2d/test/test_features2d.cpp
+++ b/modules/cudafeatures2d/test/test_features2d.cpp
@@ -55,6 +55,8 @@ namespace
     IMPLEMENT_PARAM_CLASS(FAST_NonmaxSuppression, bool)
 }
 
+namespace {
+
 PARAM_TEST_CASE(FAST, cv::cuda::DeviceInfo, FAST_Threshold, FAST_NonmaxSuppression)
 {
     cv::cuda::DeviceInfo devInfo;
@@ -707,5 +709,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_Features2D, BruteForceMatcher, testing::Combine(
     testing::Values(NormCode(cv::NORM_L1), NormCode(cv::NORM_L2)),
     testing::Values(DescriptorSize(57), DescriptorSize(64), DescriptorSize(83), DescriptorSize(128), DescriptorSize(179), DescriptorSize(256), DescriptorSize(304)),
     testing::Values(UseMask(false), UseMask(true))));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudafilters/test/test_filters.cpp
+++ b/modules/cudafilters/test/test_filters.cpp
@@ -68,6 +68,8 @@ namespace
     }
 }
 
+namespace {
+
 /////////////////////////////////////////////////////////////////////////////////////////////////
 // Blur
 
@@ -707,5 +709,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_Filters, Median, testing::Combine(
                     KernelSize(15)),
     WHOLE_SUBMAT)
     );
+
+} //namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudaimgproc/test/test_canny.cpp
+++ b/modules/cudaimgproc/test/test_canny.cpp
@@ -55,6 +55,8 @@ namespace
     IMPLEMENT_PARAM_CLASS(L2gradient, bool)
 }
 
+namespace {
+
 PARAM_TEST_CASE(Canny, cv::cuda::DeviceInfo, AppertureSize, L2gradient, UseRoi)
 {
     cv::cuda::DeviceInfo devInfo;
@@ -97,5 +99,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_ImgProc, Canny, testing::Combine(
     testing::Values(AppertureSize(3), AppertureSize(5)),
     testing::Values(L2gradient(false), L2gradient(true)),
     WHOLE_SUBMAT));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudaimgproc/test/test_histogram.cpp
+++ b/modules/cudaimgproc/test/test_histogram.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
 // HistEven
 
@@ -211,5 +213,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_ImgProc, CLAHE, testing::Combine(
     ALL_DEVICES,
     DIFFERENT_SIZES,
     testing::Values(0.0, 40.0)));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudaimgproc/test/test_hough.cpp
+++ b/modules/cudaimgproc/test/test_hough.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
 // HoughLines
 
@@ -255,5 +257,7 @@ CUDA_TEST_P(GeneralizedHough, Ballard)
 INSTANTIATE_TEST_CASE_P(CUDA_ImgProc, GeneralizedHough, testing::Combine(
     ALL_DEVICES,
     WHOLE_SUBMAT));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudaobjdetect/test/test_objdetect.cpp
+++ b/modules/cudaobjdetect/test/test_objdetect.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 //#define DUMP
 
 struct HOG : testing::TestWithParam<cv::cuda::DeviceInfo>
@@ -557,5 +559,7 @@ CUDA_TEST_P(LBP_classify, Accuracy)
 
 INSTANTIATE_TEST_CASE_P(CUDA_ObjDetect, LBP_classify,
                         testing::Combine(ALL_DEVICES, testing::Values<int>(0)));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudaoptflow/test/test_optflow.cpp
+++ b/modules/cudaoptflow/test/test_optflow.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 //////////////////////////////////////////////////////
 // BroxOpticalFlow
 
@@ -400,5 +402,7 @@ CUDA_TEST_P(OpticalFlowDual_TVL1, Accuracy)
 INSTANTIATE_TEST_CASE_P(CUDA_OptFlow, OpticalFlowDual_TVL1, testing::Combine(
     ALL_DEVICES,
     testing::Values(Gamma(0.0), Gamma(1.0))));
+
+} // namespace
 
 #endif // HAVE_CUDA

--- a/modules/cudastereo/test/test_stereo.cpp
+++ b/modules/cudastereo/test/test_stereo.cpp
@@ -46,6 +46,8 @@
 
 using namespace cvtest;
 
+namespace {
+
 //////////////////////////////////////////////////////////////////////////
 // StereoBM
 
@@ -208,5 +210,7 @@ INSTANTIATE_TEST_CASE_P(CUDA_Stereo, ReprojectImageTo3D, testing::Combine(
     DIFFERENT_SIZES,
     testing::Values(MatDepth(CV_8U), MatDepth(CV_16S)),
     WHOLE_SUBMAT));
+
+} // namespace
 
 #endif // HAVE_CUDA

